### PR TITLE
PP-503: Update PTL test skip wrapper function for cray platform

### PR DESCRIPTION
**Pull request by saritakh**
#### Issue-ID

* **[PP-503](https://pbspro.atlassian.net/browse/PP-503)**

#### Problem description
* Update PTL test skip wrapper function for cray platform

#### Cause / Analysis
* Current PTL test skip wrapper function for cray platform looks for different message which is not present

#### Solution description
* Updated the skip wrapper code to check for mom log message with "alps_client" in order to determine if the PBS build is built for cray platform
* Updated skip wrapper in few test cases of smoke test suite since job suspension is not supported on Cray right now
* Also fixed test issues for few test cases which need 1 ncpus exculsively

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by bhroam** [comment]
I'm happy with the code to solve your immediate problem.  Do note that this fix could cause future problems for the cray automation project.  Suspend resume is supported on a real cray, and this decorator will cause these tests to be skipped.  This is future work, so it shouldn't get in the way of this pull request.

Another point of confusion I had: the skip on cray decorator doesn't mean skip on a real cray.  It means skip on a cray build.  Consider renaming it to SkipOnCrayBuild.   Once again, renaming the decorator is scope creep, so maybe file a bug to do so.

File: test/fw/ptl/lib/pbs_testlib.py [hunk 1]
```diff
@@ -11547,6 +11537,16 @@ def load_configuration(self, infile):
         " load configuration from saved file infile "
         self._load_configuration(infile, MGR_OBJ_NODE)
 
+    def is_cray(self):
+        """
+        Returns True if the version of PBS used was built for Cray platforms
+        Checks built in resource vntype being set on a node
```
Comment (hirenvadalia) [inline test/fw/ptl/lib/pbs_testlib.py]: there is a mismatch in code and docstring.

**Comment by hirenvadalia** [comment]
@saritakh changes looks good. I sign-off.

**Comment by saritakh** [comment]
@bhroam @hirenvadalia Thanks.
@bhroam I have logged below Jira for renaming the decorator:
PP-519: Rename skinOnCray decorator function to indicate skip on cray build

**Comment by saritakh** [comment]
I have rebased my branch and hence request maintainers to take this forward.

**Comment by agrawalravi90** [comment]
this was urgent so I merged this in.

**Pull request merged**